contrib/endpoints/src/api_manager/service_management_fetch.cc

```diff
@@ -26,13 +26,38 @@ const int kInceptionFetchTimeout = 1000;
 // Maximum number of retries to fetch metadata
 const int kInceptionFetchRetries = 5;
 
+// Default service management API url
+const char kServiceManagementHost[] =
+    "https://servicemanagement.googleapis.com";
+
+const char kServiceManagementPath[] =
+    "/google.api.servicemanagement.v1.ServiceManager";
+
 // HTTP request callback
 typedef std::function<void(const utils::Status&, const std::string&)>
     HttpCallbackFunction;
 
+std::string get_service_management_host(
+    std::shared_ptr<context::GlobalContext> context) {
+  std::string host(kServiceManagementHost);
+
+  if (context->server_config()->has_service_management_config()) {
+    if (!context->server_config()->service_management_config().url().empty()) {
+      host = context->server_config()->service_management_config().url();
+    }
+  }
+
+  return host;
+}
+
 const std::string& get_auth_token(
     std::shared_ptr<context::GlobalContext> context) {
   if (context->service_account_token()) {
+    // register auth token for servicemanagement services
+    context->service_account_token()->SetAudience(
```

Comment:
My suggestion is: make service_management_fetch a class, and created and owned by config_manager.  In this way,  it can init and store some variables.